lib/scope_file_access.rb

```diff
@@ -64,4 +71,31 @@ def tarball_path(system_file)
       File.join(system_file.scope.scope_file_store.path, "files.tgz")
     end
   end
+
+  def retrieve_file_content(filename)
+    if File.exist?(File.join(scope_file_store.path, filename))
+      return "is here"
+    else
+      begin
+        tar_file = File.join(scope_file_store.path, "files.tgz")
+        Cheetah.run("tar", "xfO", tar_file, filename.gsub(/^\//,''), :stdout => :capture)
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.